src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -180,14 +174,23 @@ protected override DbParameterCollection DbParameterCollection
         public override UpdateRowSource UpdatedRowSource { get; set; }
 
         /// <summary>
+        /// Gets or sets the data reader currently being used by the command, or null if none.
+        /// </summary>
+        /// <value>The data reader currently being used by the command.</value>
+        protected internal virtual SqliteDataReader DataReader { get; set; }
+
+        /// <summary>
         /// Releases any resources used by the connection and closes it.
         /// </summary>
         /// <param name="disposing">
         /// true to release managed and unmanaged resources; false to release only unmanaged resources.
         /// </param>
         protected override void Dispose(bool disposing)
         {
-            DisposePreparedStatements();
+            if (disposing)
```

Comment:
I removed this condition before in order to enable the base finalizer to dispose the prepared statements.